### ISSUE TITLE
eventlet.queue.Waiter.switch() catch statement too broad

### ISSUE THREAD

**Issue by DeTeReR** [eventlet/eventlet#407]
I *think* that when a greenlet is woken by the arrival of a value in a queue the except in eventlet.queue.Waiter.switch() is too broad. The behaviour is inconsistent with what happens if the thread is already running when it 'get's from the queue.

Code to reproduce:
```
import sys
import eventlet

q = eventlet.Queue()

def get():
	print("get")
	item = q.get()
	print("Got: %s" % item)
	sys.exit(1)
eventlet.spawn(get)
eventlet.sleep()

q.put(1)

eventlet.sleep(1)
print("We should not see this.")


def put_and_get():
	q.put(1)
	get()

eventlet.spawn(put_and_get)
eventlet.sleep()

print("We don't actually see this.")
```

Should:
```
try:
  self.greenlet.switch(value)
except:
  traceback.print_exc()
```
be this instead?
```
try:
  self.greenlet.switch(value)
except Exception:
  traceback.print_exc()
```

**Comment by temoto** [comment]
Definitely agree, bare `except:` is almost always an error, true for this case too. Thank you.

**Comment by jstasiak** [comment]
Yup, there's a flake8 plugin to cover this too: https://github.com/elijahandrews/flake8-blind-except

**Comment by temoto** [comment]
Thanks Jakub, haven't seen you for a while. Is everything fine?

**Comment by temoto** [comment]
@DeTeReR fix is merged in master 0daf3e9da421088e15fb4540302af55a84b8eaf1
Have a nice life. 👍

**Comment by jstasiak** [comment]
@temoto Yeah, just following different priorities these days ;)
